crypto/hpke/hpke_highlevel.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* An OpenSSL-based high level interface to HPKE */
+#include <stddef.h>
+#include <openssl/hpke.h>
+#include <openssl/core_names.h>
+#include <openssl/crypto.h>
+#include <internal/packet.h>
+#include <openssl/err.h>
+
+int OSSL_HPKE_seal_base(OSSL_HPKE_SUITE suite,
+						const unsigned char *receiver, size_t receiverlen,
+						const unsigned char *pt, size_t ptlen,
+						const unsigned char *aad, size_t aadlen,
+						unsigned char **box, size_t *boxlen,
+						OSSL_LIB_CTX *libctx, const char *propq)
+{
+    size_t ctlen;
+    size_t enclen;
+    unsigned char *enc = NULL;
+    unsigned char *ct = NULL;
+    OSSL_HPKE_CTX *ctx = NULL;
+    WPACKET pkt;
+    int ret = 0;
+
+    if (box == NULL || *box != NULL || boxlen == NULL) {
+        ERR_raise(ERR_LIB_CRYPTO, ERR_R_PASSED_INVALID_ARGUMENT);
+        goto end;
```

Comment:
The first 3 gotos here can just be return 0

Reply:
On the other hand, this makes for consistent code in this function